Sources/CoreAardvark/PrivateCategories/NSFileHandle+ARKAdditions.m

```diff
@@ -42,10 +42,15 @@ - (void)ARK_writeDataBlock:(NSData *)dataBlock;
     uint8_t dataLengthBytes[ARKBlockLengthBytes] = { };
     ARKWriteBigEndianBlockLength(dataLengthBytes, 0, dataBlockLength);
     NSData *dataLengthData = [NSData dataWithBytes:dataLengthBytes length:ARKBlockLengthBytes];
-    
+
+    NSMutableData *combinedData = [dataLengthData mutableCopy];
+    [combinedData appendData:dataBlock];
+
+    __weak NSData *weakCombinedData = combinedData;
```

Comment:
Did you test this with optimizations enabled? I think this may not work. ARC is allowed to release `combinedData` right after this line, and `weakCombinedData` would then be nil.

Reply:
only ran tests in debug mode

since `weakCombinedData` is referenced below in a nested scope, i don't think ARC would relase things immediately after this line?

Reply:
It would because it's a weak reference. Weak references don't extend their lifetime. In fact, you get a warning in Swift doing this precisely for that reason:

```swift
weak var bar = NSObject()
print(bar) // This actually prints nil!
```

Reply:
i think the swift equivalent here would be something like

```
        let x = NSObject()
        weak var weakX = x
        autoreleasepool {
            _ = weakX
        }
```

which does not result in any warnings!